### ISSUE TITLE
Ignore UNT0039 on generics types

### ISSUE THREAD

**Issue by amedinae** [microsoft/Microsoft.Unity.Analyzers#445]
#### Bug description
The analyzer incorrectly triggers UNT0039 ("Use RequireComponent attribute when self-invoking GetComponent") when GetComponent<T>() is called using a generic type parameter rather than a concrete type.

- Version of analyzers assembly: 1.26.0
- Analyzer rule: UNT0039
- Error (exception message, type, and callstack where applicable): False positive diagnostic

#### To Reproduce
```
using UnityEngine;

public class PlayerScript : MonoBehaviour
{
    void Foo<T>()
    {
        var bar = GetComponent<T>();
    }
}
```

#### Expected behavior
The analyzer should ignore GetComponent<T>() calls where T is a generic type parameter. It should only trigger when T is a concrete type that can validly be used in a [RequireComponent] attribute.
